Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfileController.cs

```diff
@@ -163,22 +164,21 @@ public async Task<ActionResult> UpdateAsync([FromBody] ProfileApiModel apiModel)
             }
         }
 
-        [HttpPost("follow/{user=user}")]
+        [HttpPost("follow/{followEmail=email}")]
```

Comment:
that's not how the syntax works. `=` sets a default value, making the parameter optional. `:` sets a [constraint](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-5.0#route-constraint-reference), but there is no email constraint. anyway, we don't need to handle that at routing, it's really the job of model validation. you can validate email address format if you want with [EmailAddress], but the utility is low

we don't need the check for if email is null then 400, and we don't need the RequiredAttribute either, since if you make the route template "follow/{followEmail}" then it's a required parameter & there's no way a null or empty string could ever be routed to the method

Reply:
Removed the default value on the route and removed redundant null checks on the route parameter. Also changed RequiredAttribute to EmailAddress.